File: recovered/src/com/lumiyaviewer/lumiya/ui/settings/GoogleDriveSyncPreference.java
package com.lumiyaviewer.lumiya.ui.settings;

import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.net.Uri;
import android.support.v7.app.AlertDialog;
import android.support.v7.preference.CheckBoxPreference;
import android.util.AttributeSet;
import com.lumiyaviewer.lumiya.R;
import com.lumiyaviewer.lumiya.licensing.LicenseChecker;
import com.lumiyaviewer.lumiya.sync.CloudSyncServiceConnection;

/* loaded from: classes.dex */
public class GoogleDriveSyncPreference extends CheckBoxPreference {
    public GoogleDriveSyncPreference(Context context) {
        super(context);
    }

    public GoogleDriveSyncPreference(Context context, AttributeSet attributeSet) {
        super(context, attributeSet);
    }

    public GoogleDriveSyncPreference(Context context, AttributeSet attributeSet, int i) {
        super(context, attributeSet, i);
    }

    public GoogleDriveSyncPreference(Context context, AttributeSet attributeSet, int i, int i2) {
        super(context, attributeSet, i, i2);
    }

    /* renamed from: lambda$-com_lumiyaviewer_lumiya_ui_settings_GoogleDriveSyncPreference_1541, reason: not valid java name */
    /* synthetic */ void m863x92031b8a(DialogInterface dialogInterface, int i) {
        dialogInterface.dismiss();
        Intent intent = new Intent("android.intent.action.VIEW");
        intent.setData(Uri.parse(LicenseChecker.CLOUD_PLUGIN_URL));
        getContext().startActivity(intent);
    }

    @Override // android.support.v7.preference.TwoStatePreference, android.support.v7.preference.Preference
    protected void onClick() {
        super.onClick();
        if (!isChecked() || CloudSyncServiceConnection.checkPluginInstalled(getContext())) {
            return;
        }
        new AlertDialog.Builder(getContext()).setTitle(R.string.enable_google_drive).setMessage(getContext().getString(R.string.enable_google_drive_message, LicenseChecker.APP_STORE_NAME)).setPositiveButton("Yes", new DialogInterface.OnClickListener() { // from class: com.lumiyaviewer.lumiya.ui.settings.-$Lambda$GONhG2H9-w043w0Zbd-p0nmAUgQ.1
            private final /* synthetic */ void $m$0(DialogInterface dialogInterface, int i) {
                ((GoogleDriveSyncPreference) this).m863x92031b8a(dialogInterface, i);
            }

            @Override // android.content.DialogInterface.OnClickListener
            public final void onClick(DialogInterface dialogInterface, int i) {
                $m$0(dialogInterface, i);
            }
        }).setNegativeButton("No", new DialogInterface.OnClickListener() { // from class: com.lumiyaviewer.lumiya.ui.settings.-$Lambda$GONhG2H9-w043w0Zbd-p0nmAUgQ
            private final /* synthetic */ void $m$0(DialogInterface dialogInterface, int i) {
                dialogInterface.cancel();
            }

            @Override // android.content.DialogInterface.OnClickListener
            public final void onClick(DialogInterface dialogInterface, int i) {
                $m$0(dialogInterface, i);
            }
        }).setCancelable(true).create().show();
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/ui/settings/GoogleDriveSyncPreference.kt
package com.lumiyaviewer.lumiya.ui.settings

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.util.AttributeSet
import androidx.appcompat.app.AlertDialog
import androidx.preference.CheckBoxPreference
import com.lumiyaviewer.lumiya.R
import com.lumiyaviewer.lumiya.licensing.LicenseChecker
import com.lumiyaviewer.lumiya.sync.CloudSyncServiceConnection

class GoogleDriveSyncPreference : CheckBoxPreference {
    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) :
        super(context, attrs, defStyleAttr, defStyleRes)

    override fun onClick() {
        super.onClick()
        if (!isChecked || CloudSyncServiceConnection.checkPluginInstalled(context)) {
            return
        }
        AlertDialog.Builder(context)
            .setTitle(R.string.enable_google_drive)
            .setMessage(context.getString(R.string.enable_google_drive_message, LicenseChecker.APP_STORE_NAME))
            .setPositiveButton("Yes") { dialog, _ ->
                dialog.dismiss()
                val intent = Intent(Intent.ACTION_VIEW)
                intent.data = Uri.parse(LicenseChecker.CLOUD_PLUGIN_URL)
                context.startActivity(intent)
            }
            .setNegativeButton("No") { dialog, _ -> dialog.cancel() }
            .setCancelable(true)
            .create()
            .show()
    }
}
